vm/interpreter/src/vm.rs

```diff
@@ -397,12 +434,88 @@ where
     }
 }
 
+#[derive(Clone, Default)]
+struct GasOutputs {
+    pub base_fee_burn: TokenAmount,
+    pub over_estimation_burn: TokenAmount,
+    pub miner_penalty: TokenAmount,
+    pub miner_tip: TokenAmount,
+    pub refund: TokenAmount,
+
+    pub gas_refund: i64,
+    pub gas_burned: i64,
+}
+
+fn compute_gas_outputs(
+    gas_used: i64,
+    gas_limit: i64,
+    base_fee: &TokenAmount,
+    fee_cap: &TokenAmount,
+    gas_premium: TokenAmount,
+) -> GasOutputs {
+    let gas_used_big: BigInt = 0.into();
+    let mut base_fee_to_pay = base_fee;
+    let mut out = GasOutputs::default();
```

Comment:
Opinionated but seems like it would be a bit cleaner to construct all variables, then just construct the `GasOutputs` type for the return